Changelog.txt

```diff
@@ -3429,9 +3429,219 @@ Additionally, the problem of zig-zag issue following in the South direction has
+	ARGO = the item that drank
+	ARGN1 = cooldown (in seconds)
+	ARGN2 = consumption amount (1 in default)
+	LOCAL.BottleID = empty bottle id that will be given when after item drank. (Will be given ARGN2 amount of empty bottle in default)
+	- Returns:
+	1: Stops character from drinking the item.
+	5: Let character to drink but will only give 1 empty bottle.
+	6: Let character to drink but will not give any empty bottle.
+	- Messages:
+	DRINK_NOT_ENOUGH, "You don't have enough %s to drink this."
+- Added: New trigger @AfkMode added for characters. (Issue: #1161)
+	I: The character changing AFK status.
+	SRC: The character changing AFK status.
+	ARGN1 = Current Afk Status
+	ARGN2 = Next Afk Status
+	- Returns:
+	1: Stops character from switching afk status.
+- Added: New trigger @Reveal added for characters. (Issue: #1157)
+	I: The character who is revealing.
+	SRC: The character who is revealing.
+	- Returns:
+	1: Prevents character from being revealed.
+- Added: New triggers @ArrowQuest_Add and @ArrowQuest_Close added for characters. (Issue: #1116)
+	I: The character who activating/deactivating arrow quest.
+	SRC: The character who activating/deactivating arrow quest.
+- Added: New function triggers f_onaccount_block and f_onaccount_unblock. (Issue: #1156)
+	ARGS = Account name.
+	- Returns:
+	1: Prevents account from blocking/unblocking.
+- Added: New function trigger f_onserver_broadcast. (Issue: #1145)
+	ARGS = message
+	- Returns:
+	1: Prevents server to send broadcast message.
+- Added: Two new LOCAL variables ItemPoisonReductionChance and ItemPoisonReductionAmount to @Hit trigger. (Issue: #1159)
+	This LOCALs changes default behaviour of poisoned weapons while removing poison.
+	LOCAL.ItemPoisonReductionChance = Is the chance that weapon reduce poison morez. (Default 100%)
+	LOCAL.ItemPoisonReductionAmount = Is the amount of poison that weapon reduce. (Default: CurrentPoisonAmount/ 2)
+- Added: You can now use TDATA2 for setting default cooldown (delay) for drinks. (Default: 15secs for potions, 1500 secs for ales)
+- Changed: Reactive armor range increased from 1 to 2 in default.
+
+27-03-2024, xwerswoodx
+- Fixed: Possible fix for accesses and bans not loading for multis. (Issue: #1223)
+- Fixed: Possible fix for wrong damage calculation of slayer bonus damage. (Issue: #1179)
+
+28-03-2024, xwerswoodx
+- Added: REGION_FLAG_HOUSE (0x0100) for houses. (Issue: #1154)
+	REGION_FLAG_HOUSE 0100 should be added defs.
+
+30-03-2024, xwerswoodx
+- Added: ARGO object for @MurderMark trigger to get killed character.
+- Fixed: ARGN1 (reap amount) doesn't updated depends on the amount left on bit. (Issue: #1224)
+- Added: ARGO to get bit uid under @ResourceGather and @RegionResourceGather triggers.
+
+31-03-2024, xwerswoodx
+- Fixed: Possible fix for the OnRegionTrigger getting called on invalid region while multi creating/removing.
+
+02-04-2024, xwerswoodx
+- Changed: dword CAN flags changed with uint64 to increase limits.
```

Comment:
Honestly I was pretty sure it's was working fine, and I checked it on CANMASK and it seems like working fine and max value is 0xFFFF FFFF FFFF FFFF. But eval cannot evaluate it seems like EVAL need to be updated for uint64 values if needed, maybe because of evaluated expressions it doesn't work for if checks, I need to check more deeply.

Reply:
Some stuff internally is managed as unsigned values to be more clear about the usage and to remove sign management. You could read/print out a uint64 max value as text (even if it shouldn't happen) while executing scripts, but when you set a property or whatever to such a number via scripts most often you will get a wrong number, because scripts manage most numbers as signed ones. There might be some exception to this rule, but i wouldn't expect uint64 max val to work everywhere (we also had similar bugs in the past afair)

Reply:
I thought same so I put some information in changelog to not use values more than 2 << 47 ( 0x1 0000 0000 0000) which is the last flag can be used for int64.

Reply:
Afaik the last flag should be binary 0100 (and other 15 0000 groups), so 1 << 62, or 4000 0000 0000 0000 hex. We're leaving the upper bit (64th) for the sign

Reply:
Yes but I suggest them to not go more than << 47 to make sure eval, qval, hval works fine with them, of course they could go up to << 63 but there could be issues if they don't know what they are doing. Honestly I don't think anyone gonna add 16+ new can flags in script side, they may add 1-2 new flags. If you wish I could add max flag value as << 62 for changelog, but if they are not knowing about bits, they could probably have many evaluation issues. So for me it's best to tell them to not use more than << 47 to make sure the values are in evaluation range. Of course this is just my own thought.

Reply:
Why 48 bits as a magic number? I agree to warn about fiddling with it only if expert, but i'd give the correct informations on the max val instead of only "safe" informations

Reply:
Okay, I thought experts should know what's the max value of uint64 so no need to inform about it, but I updated the change log for it.